ckan/config/middleware/flask_app.py

```diff
@@ -340,6 +304,16 @@ def get_locale() -> str:
         config.get_value(u'ckan.locale_default'))
 
 
+def remote_user() -> None:
+    header = "REMOTE_USER"
+
+    if "_user_id" in request.environ.get("beaker.session", ""):
+        user_id = request.environ["beaker.session"]["_user_id"]
+        userobj = model.Session.query(model.User).get(user_id)
+        if userobj:
+            request.environ[header] = userobj.name
```

Comment:
This function is for backwards compatibility. Can we add some docstring to document:
- Why it is needed here
- If we can remove in the future and conditions for it.

Reply:
Thanks for your notes @pdelboca  i really appreciate it!

First i will try to explain why at first i thought we needed function like this.

We needed it to set the `g.user` to the current logged user which we are getting from our `beaker_session['_user_id']`  not because we only need it for backward compatibility for the extensions like `saml2auth`, but everywhere in `CKAN` we are dependant from `g.user/g.userobj`. 

Even if we log in the user using `login_user` method from flask, our `CKAN` is not recognizing it if i comment out this line of code: 
https://github.com/ckan/ckan/blob/71a82c4b0bb499fd3a6d1ccfd038b2231f50f92a/ckan/views/__init__.py#L170
because as i said we are dependant from `g.user` and `g.userobj` everywhere in our app, thats why i made this function in the first place.

But  now as I write this, i think i come up with a better solution: 

To make this to be more in a `Flask` way, we should use the `current_user` proxy, and instead of `REMOTE_USER`, `current_user` will set the g.user and g.userobj and with this, we will remove our dependancy from `REMOTE_USER`

We could remove our dependency even from g.user/g.userobj (we can access the logged-in user with the [current_user](https://flask-login.readthedocs.io/en/latest/#flask_login.current_user) proxy, which is available in every template:)

e.g
```python
{% if current_user.is_authenticated %}
  {{ current_user.name }}!
{% endif %}
```


Thanks